drivers/si1133/si1133.c

```diff
@@ -0,0 +1,353 @@
+            return SI1133_ERR_LOGIC;
+        }
+        /* The reset command is done when the RUNNING flag is clear, the other
+         * commands are done when the command value is set to the expected
+         * one. */
+        if ((command == SI1133_CMD_RESET_SW)
+            ? !(new_cmd_ctr & SI1133_RESP0_RUNNING_MASK)
+            : dev->cmd_counter
+            == (new_cmd_ctr & SI1133_RESP0_COUNTER_MASK)) {
+            break;
+        }
+        /* The command didn't yet finish in this case so it should be in running
+         * state and we need to retry the loop with a timeout. This avoids
+         * calling xtimer for commands that are immediate. */
+        if (retry) {
+            if (xtimer_usec_from_ticks(xtimer_diff(xtimer_now(), start_time))
+                > SI1133_COMMAND_TIMEOUT_USEC) {
+                DEBUG("[si1133] Command 0x%.2x timeout.\n", (unsigned)command);
+                return SI1133_ERR_LOGIC;
+            }
+        }
+        else {
+            retry = true;
+            start_time = xtimer_now();
+        }
+    }
+    return SI1133_OK;
+}
+
+static int _si1133_set_param(si1133_t *dev, uint8_t param, uint8_t value)
+{
+    int ret;
+
+    ret = i2c_write_reg(dev->i2c_dev, dev->address,
+                        SI1133_REG_HOSTIN0, value, 0 /* flags */);
+    if (ret) {
+        DEBUG("[si1133] write HOSTIN0 I2C error: %s\n", strerror(-ret));
+        return SI1133_ERR_I2C;
+    }
+    ret = _si1133_run_command(dev, param | SI1133_CMD_PARAM_SET);
+    if (ret) {
+        return ret;
+    }
+
+    uint8_t resp1;
+    ret = i2c_read_reg(dev->i2c_dev, dev->address,
+                       SI1133_REG_RESPONSE1, &resp1, 0);
+    if (ret) {
+        DEBUG("[si1133] read RESPONSE1 I2C error: %s\n", strerror(-ret));
+        return SI1133_ERR_I2C;
+    }
+    if (resp1 != value) {
+        DEBUG("[si1133] Expected to read back value 0x%.2" PRIu8
+              " when setting param 0x%.2" PRIu8 " but got 0x%.2" PRIu8 "\n",
+              value, param, resp1);
+        return SI1133_ERR_LOGIC;
+    }
+    return SI1133_OK;
+}
+
+/**
+ * @brief Reset the device.
+ */
+static int _si1133_reset(si1133_t *dev)
+{
+    DEBUG("[si1133] reset()\n");
+    dev->sensors_mask = 0;
+    int ret = _si1133_run_command(dev, SI1133_CMD_RESET_SW);
+    if (ret) {
+        return ret;
+    }
+
+    return _si1133_run_command(dev, SI1133_CMD_RESET_CMD_CTR);
+}
+
+static int _si1133_configure(si1133_t *dev, uint8_t sensors_mask)
+{
+    DEBUG("[si1133] configure(0x%.2x)\n", (unsigned)sensors_mask);
+    if (sensors_mask == dev->sensors_mask) {
+        /* Nothing to do if it is already configured. */
+        return SI1133_OK;
+    }
+
+    int ret;
+
+    dev->sensors_mask = sensors_mask;
+    uint8_t chan_list = 0;
+    uint8_t mask = sensors_mask;
+    for (uint8_t chan = 0; mask && chan < SI1133_NUM_CHANNELS;
+         chan++, mask = mask & (mask - 1)) {
+        si1133_sensor_t sensor = mask ^ (mask & (mask - 1));
```

Comment:
I have the feeling that `bitarithm_test_and_clear()` might again be useful here

Reply:
`bitarithm_test_and_clear` doesn't define the order in which you process the bits. Also, I don't need the index of the bits, only need the 1-bit mask.